### ISSUE TITLE
[Package]:NixOS Flake 

### ISSUE THREAD

**Issue by Karitham** [ThaUnknown/miru#408]
### Preflight checklist

- [X] I have searched the [issue tracker](https://github.com/ThaUnknown/miru/issues) for a bug report that matches the one I want to file, without success.
- [X] I have searched the [features list](https://github.com/ThaUnknown/miru#features) for this feature, and I couldn't find it.
- [X] I have checked that I'm using the [latest stable](https://github.com/ThaUnknown/miru/releases/latest) version of the app.

### Problem Description

It's a bit of a pain to install miru on nixos. There's two options:

- Install the deb file manually, fix all the issues (horrible)
- Install the AppImage, use external appImage runner

### Proposed Solution

Integrate a flake to the repo to use as a package. This would benefit from auto-updating directly from the repo's CI.

I already have a working derivation of miru (<https://github.com/Karitham/knixpkgs/blob/master/miru.nix>), just wondering if you would approve of it being directly part of the repo instead of in an external repository.

This would transform the hell that is to install miru on any linux distro that's not ubuntu or arch
to just

```sh
$ nix profile install github:thaunknown/miru
```

Let me know if you're interested in a PR

**Comment by ThaUnknown** [comment]
I have low interest in supporting linux package managers, because there are more issues per package manager, than in all of Miru together, to put it simply it's a waste of time

I distribute files which I know work, that is AppImage and deb, and if something doesn't work that's the end user's fault, general rule is "you use linux, deal with linux issues yourself", as the linux userbase is way too small to warrant wasting 50% of my time to fix each and small issue

I'll accept PR's for CI which automatically distributes Miru to package managers, [like the winget CI](https://github.com/ThaUnknown/miru/blob/master/.github/workflows/winget.yml), but if it causes issues with how Miru functions I'll likely remove it later on

**Comment by ThaUnknown** [comment]
what's the status on this?

**Comment by Karitham** [comment]
TLDR: my opinion is dont add the flake, maybe document it exists somewhere, end of story

> I have low interest in supporting linux package managers, because there are more issues per package manager, than in all of Miru together, to put it simply it's a waste of time

Considering there is *some* maintenance overhead for supporting nix (largely adding deps when needed etc), the overhead to maintaining the flake on this repo is higher than having the user fetch the package from my flake above.

Upgrading the version themselves is also easy — in nix you can pick a package, overlay a new version and have it rebuilt in 3 lines, I propose not adding the flake to this repo and maybe just directing nix users to a maintained flake. (Mine should be).

If you don't want to point users to third parties for packaging that's also fine, people can look it up.

**Comment by cloudglides** [comment]
bit late
but here you go
https://github.com/cloudglides/hayase-nix

**Comment by cloudglides** [comment]
Aye aye sir, imma fix em soon
